wings/market/coinbase_pro_market.pyx

```diff
@@ -0,0 +1,816 @@
+        super().__init__()
+        self._coinbase_auth = CoinbaseProAuth(coinbase_pro_api_key, coinbase_pro_secret_key, coinbase_pro_passphrase)
+        self._order_book_tracker = CoinbaseProOrderBookTracker(data_source_type=order_book_tracker_data_source_type,
+                                                               symbols=symbols)
+        self._account_balances = {}
+        self._ev_loop = asyncio.get_event_loop()
+        self._poll_notifier = asyncio.Event()
+        self._last_timestamp = 0
+        self._poll_interval = poll_interval
+        self._in_flight_deposits = {}
+        self._in_flight_orders = {}
+        self._tx_tracker = CoinbaseProMarketTransactionTracker(self)
+        self._w3 = Web3(Web3.HTTPProvider(web3_url))
+        self._trading_rules = {}
+        self._data_source_type = order_book_tracker_data_source_type
+        self._status_polling_task = None
+        self._order_tracker_task = None
+        self._shared_client = None
+
+    @property
+    def order_books(self) -> Dict[str, OrderBook]:
+        return self._order_book_tracker.order_books
+
+    @property
+    def coinbase_auth(self) -> CoinbaseProAuth:
+        return self._coinbase_auth
+
+    @property
+    def ready(self) -> bool:
+        return (len(self._order_book_tracker.order_books) > 0 and
+                len(self._account_balances) > 0 and
+                len(self._trading_rules) > 0)
+
+    def get_all_balances(self) -> Dict[str, float]:
+        return self._account_balances.copy()
+
+    cdef c_start(self, Clock clock, double timestamp):
+        self._tx_tracker.c_start(clock, timestamp)
+        MarketBase.c_start(self, clock, timestamp)
+        self._order_tracker_task = asyncio.ensure_future(self._order_book_tracker.start())
+        self._status_polling_task = asyncio.ensure_future(self._status_polling_loop())
+
+    cdef c_tick(self, double timestamp):
+        cdef:
+            int64_t last_tick = <int64_t>(self._last_timestamp / self._poll_interval)
+            int64_t current_tick = <int64_t>(timestamp / self._poll_interval)
+
+        MarketBase.c_tick(self, timestamp)
+        if current_tick > last_tick:
+            if not self._poll_notifier.is_set():
+                self._poll_notifier.set()
+        self._last_timestamp = timestamp
+
+    async def _http_client(self) -> aiohttp.ClientSession:
+        if self._shared_client is None:
+            self._shared_client = aiohttp.ClientSession()
+        return self._shared_client
+
+    async def _api_request(self,
+                           http_method: str,
+                           path_url: str = None,
+                           url: str = None,
+                           data: Optional[Dict[str, any]] = None) -> Dict[str, any]:
+        assert path_url is not None or url is not None
+
+        url = f"{self.COINBASE_API_ENDPOINT}{path_url}" if url is None else url
+        data_str = "" if data is None else json.dumps(data)
```

Comment:
minor - we use  ujson in other files

Reply:
I believe we use both. Leaving it for now.